documentation/examples/remote_storage/remote_storage_adapter/main.go

```diff
@@ -296,7 +296,7 @@ func serve(logger log.Logger, addr string, writers []writer, readers []reader) e
 
 		compressed = snappy.Encode(nil, data)
 		if _, err := w.Write(compressed); err != nil {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
+			level.Warn(logger).Log("msg", "Resonse write error", "storage", reader.Name(), "err", err)
```

Comment:
It would be a better fix to move L294-295 after this block.

Reply:
The error is not isolated to L294-295.
 Function [w.Write](https://github.com/golang/go/blob/889aa5eb98ff2a5134f5b075525479da399e5c4f/src/net/http/server.go#L367) will WriteHeader and function [http.Error](https://github.com/golang/go/blob/889aa5eb98ff2a5134f5b075525479da399e5c4f/src/net/http/server.go#L2004) will WriteHeader too.
Function Write
```golang
func (cw *chunkWriter) Write(p []byte) (n int, err error) {
	if !cw.wroteHeader {
		cw.writeHeader(p)
	}
	...
}
```
Function Error
```golang
func Error(w ResponseWriter, error string, code int) {
	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
	w.Header().Set("X-Content-Type-Options", "nosniff")
	w.WriteHeader(code)
	fmt.Fprintln(w, error)
}
```

Reply:
My bad, I overlooked the code. 

```go
level.Warn(logger).Log("msg", "Error writing response", "storage", reader.Name(), "err", err)
```

And you can remove `return` at L300.

Reply:
Okay, I have removed the useless return as your  suggest.

Reply:
Thanks! Can you also change `"Resonse write error"` to `"Error writing response"` please?

Reply:
My fault! Changed.